### PR TITLE
BidsCube Bid Adapter: add new adapter

### PR DESCRIPTION
**Pull request by vlaktion**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Review by msm0504 — CHANGES_REQUESTED** [review]

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/2944

File: modules/bidscubeBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { registerBidder } from '../src/adapters/bidderFactory.js'
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js'
+import * as utils from '../src/utils.js'
+
+const BIDDER_CODE = 'bidscube'
+const URL = 'https://supply.bidscube.com/?c=o&m=multi'
```
[the remark this answers is not in the document]
Comment (vlaktion) [reply modules/bidscubeBidAdapter.js:6]: We will deploy our server application after the adapter has been approved.

**Comment by vlaktion** [comment]
All done

File: modules/bidscubeBidAdapter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (vlaktion) [reply modules/bidscubeBidAdapter.js:6]: We deployed the server and it works

File: modules/bidscubeBidAdapter.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+# Test Parameters
+```
+    var adUnits = [{
+                code: 'placementId_0',
+                sizes: [[300, 250]],
+                bids: [{
+                        bidder: 'bidscube',
+                        params: {
+                            placementId: 0,
```
[the remark this answers is not in the document]
Comment (vlaktion) [reply modules/bidscubeBidAdapter.md]: We deployed the server and it works. Can you approve commits?

**Review by patmmccann — CHANGES_REQUESTED** [review]
You must return advertiserDomains

File: modules/bidscubeBidAdapter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (msm0504) [reply modules/bidscubeBidAdapter.js:6]: I'm still seeing the same error for `https://supply.bidscube.com`

[the remark this answers is not in the document]
Comment (vlaktion) [reply modules/bidscubeBidAdapter.js:6]: It needs a parameter. For example https://supply.bidscube.com/?c=o&m=multi

[the remark this answers is not in the document]
Comment (vlaktion) [reply modules/bidscubeBidAdapter.js:6]: And POST method with example payload
{
    "bidId": "9ec5b177515ee2e5",
    "bidder": "bidscube",
    "placements": [{
      "placementId": 0,
      "traffic": "banner",
      "allParams": "{}"
    }]
}

[the remark this answers is not in the document]
Comment (vlaktion) [reply modules/bidscubeBidAdapter.js:6]: You can test it throw Postman

[the remark this answers is not in the document]
Comment (msm0504) [reply modules/bidscubeBidAdapter.js:6]: @vlaktion When I try using a local test page, I'm now seeing this error:
```
Access to XMLHttpRequest at 'https://supply.bidscube.com/?c=o&m=multi' from origin 'http://localhost:3000' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header has a value 'http://localhost' that is not equal to the supplied origin.
```

[the remark this answers is not in the document]
Comment (vlaktion) [reply modules/bidscubeBidAdapter.js:6]: Fixed this problem

**Comment by patmmccann** [comment]
@vlaktion reminder that you must return advertiserDomains, see #6650 for reference

File: modules/bidscubeBidAdapter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (vlaktion) [reply modules/bidscubeBidAdapter.js:6]: Can you approve commits?

**Review by msm0504 — APPROVED** [review]

**Comment by msm0504** [comment]
@vlaktion The latest changes allowed me to get data from your server when using my test page. I've approved the PR.

However, before I merge it, please address @patmmccann's comment from last week.

**Comment by vlaktion** [comment]
@patmmccann I see the advertiserDomains is Optional. I don't have this information. Can you approve my PR?

**Comment by ChrisHuie** [comment]
@vlaktion it will be required in prebid 5.0 which is being released shortly

**Comment by vlaktion** [comment]
@patmmccann All done! For test:
POST https://supply.bidscube.com/?c=o&m=multi
{
    "bidId": "9ec5b177515ee2e5",
    "bidder": "bidscube",
    "placements": [{
      "placementId": 0,
      "traffic": "banner",
      "allParams": "{}"
    }]
}

**Comment by patmmccann** [comment]
Excellent! Can you add a unit test that demonstrates meta.advertiserDomains is set in the interpreted response?

**Comment by vlaktion** [comment]
@patmmccann Added a unit test that demonstrates meta.advertiserDomains is set in the interpreted response

**Comment by vlaktion** [comment]
Hi. Why is BidsCube adapter don`t  exist in https://docs.prebid.org/download.html?

**Comment by vlaktion** [comment]
@patmmccann Hi. Can you answer on my question?

**Pull request merged**